x/incentives/keeper/gauge.go

```diff
@@ -108,9 +109,12 @@ func (k Keeper) CreateGauge(ctx sdk.Context, isPerpetual bool, owner sdk.AccAddr
 		}
 	}
 
-	// Ensure that the denom this gauge pays out to exists on-chain
-	if !k.bk.HasSupply(ctx, distrTo.Denom) && !strings.Contains(distrTo.Denom, "osmovaloper") {
-		return 0, fmt.Errorf("denom does not exist: %s", distrTo.Denom)
+	// check if distrTo.Denom is uosmo, if not ensure that the denom this gauge
+	// pays out to exists on-chain
+	if distrTo.Denom != appParams.BaseCoinUnit {
+		if !k.bk.HasSupply(ctx, distrTo.Denom) && !strings.Contains(distrTo.Denom, "osmovaloper") {
```

Comment:
Why would the denom ever contain osmovaloper?

Reply:
yea this looks like address prefix, not sure why this check was placed here. I didn't wanted to remove the coverage